File: src/main/kotlin/dev/jefy/connectpro/user/UserClient.java
package dev.jefy.connectpro.user;

import org.jspecify.annotations.NullMarked;

import dev.jefy.connectpro.user.applicaion.dtos.UserData;
import dev.jefy.connectpro.user.domain.vo.UserId;

/**
 * @author Jôph Yamba
 */
@NullMarked
public interface UserClient {
    UserData getData(UserId userId);
    UserData getCurrentUser();

}

File: src/main/kotlin/dev/jefy/connectpro/user/UserClient.kt
package dev.jefy.connectpro.user

import dev.jefy.connectpro.user.application.dtos.UserData
import dev.jefy.connectpro.user.domain.vo.Email
import dev.jefy.connectpro.user.domain.vo.UserId
/**
 * @author  Jôph Yamba
 */
interface UserClient {
    fun getById(userId: UserId): UserData
    fun getByEmail(email: Email): UserData
    fun getCurrentUser(): UserData
}